org.adempiere.base/src/org/compiere/model/PO.java

```diff
@@ -2384,7 +2384,7 @@ private boolean saveFinish (boolean newRecord, boolean success)
 			m_createNew = false;
 		}
 		if (!newRecord) {
-			CacheMgt.get().reset(p_info.getTableName());
+			CacheMgt.get().reset(p_info.getTableName(), get_ID());
```

Comment:
Hi @hengsin - nice solving this, has been that way since compiere times.

There is another full reset on the delete method, line 3545, probably same fix?

Reply:
hi @globalqss , good catch. same fix.